### DISCUSSION TITLE
API List code scanning alerts

### DISCUSSION THREAD

**Discussion by aleks-ivanov** [community/community#24321] [Apps, API and Webhooks]
<p>When I make a request to the GitHub API for the list of code scanning alerts, I get max of 30 results.</p>
<p>Is there a way to increase that number ?</p>

**Comment by marcogario** [comment]
<p>:wave:</p>
<p>You should be able to add a <code>page</code> query parameter to your request.</p>
<p>In general, the Code Scanning Alerts endpoint is paginated according to <a href="https://docs.github.com/en/rest/overview/resources-in-the-rest-api#pagination" class="inline-onebox" rel="noopener nofollow ugc">Resources in the REST API - GitHub Docs</a></p>
<p>Let me know if you have additional questions/feedback!</p>

**Comment by aleks-ivanov** [comment]
<p>Thank you!</p>
<p>Can I use in the url query any JSON properties of the object that a GET request returns or they should be explicitly specified in the API ?</p>
<p>For example when trying to get a list of code scanning alerts and want to filter only the ones with “state”: “open”, I could do:</p>
<p>“<a href="https://api.github.com/repos/%24OWNER/%24PROJECT_NAME/code-scanning/alerts" rel="noopener nofollow ugc">https://api.github.com/repos/$OWNER/$PROJECT_NAME/code-scanning/alerts</a>?<strong>state=open</strong>”</p>
<p>“state” is an actual property of the returned object, with two possible values “open” and “dismissed”.</p>
<p>There are multiple other properties, which I might use, but am not sure if I can use them.</p>

**Comment by marcogario** [comment]
<blockquote>
<p>Can I use in the url query any JSON properties of the object that a GET request returns or they should be explicitly specified in the API ?</p>
</blockquote>
<p>Unfortunately not. Filtering by state and ref should be supported.</p>
<p>We have been working on improvements on the API, and will be releasing some updates in the next few weeks. As far as filtering Alerts goes, we are aiming at supporting filtering by tool. Feel free to share what other fields you’d find useful to filter on.</p>

**Comment by aleks-ivanov** [comment]
<p>I understand, thank you.</p>
<p>I got one more question, regarding the Code Scanning REST API.</p>
<p>For each alert that is generated, there is a full path to the file the alert originates from. When I access the API, that information is not in there.</p>
<p>Where does it come from and is it accessible by users ?</p>

**Comment by bogdanap** [comment, marked as answer]
<p>Hi <a class="mention" href="/aleksivanovsinglet">@AleksIvanovSinglet</a>!</p>
<p>I’ve replied to your question on the <a href="https://github.com/github/codeql-action/issues/383">CodeQL-action issue</a>, but to reiterate, we’re planning quite a few improvements to our API in the near future and adding this information to the responses is just one of them.</p>
<p>And since you’ve been keen on testing the current functionality and came up with good questions for us so far, we were wondering if you’d like early access to these new features?<br>
This would mean you’d get more details about the alerts (and a few new endpoints), but it also comes with the disclaimer that it is still work in progress, so bugs and potentially breaking changes can happen :slightly_smiling_face:</p>

**Comment by aleks-ivanov** [comment]
<p>Hi <a class="mention" href="/bogdanap">@bogdanap</a> ,</p>
<p>by pure cosmic chance, I read your replies in the same order you read and replied to mine, so the safety of the contextual timeline was preserved! :smiley:</p>
<p>I am really excited to hear about the improvements and also would love the opportunity for early access! Thank you!</p>
<p>Would it be possible for the early access to be granted on a GitHub Org level, so my three colleagues and I can play with the new toys, together ? :slight_smile:</p>

**Comment by bogdanap** [comment]
<aside class="quote no-group" data-username="AleksIvanovSinglet" data-post="7" data-topic="156428">
<div class="title">
<div class="quote-controls"></div>
 AleksIvanovSinglet:</div>
<blockquote>
<p>I am really excited to hear about the improvements and also would love the opportunity for early access! Thank you!</p>
</blockquote>
</aside>
<p>Yay! We’re glad to hear that! I’ll start the process then, and will come back to you with more details in a bit.</p>
<aside class="quote no-group" data-username="AleksIvanovSinglet" data-post="7" data-topic="156428">
<div class="title">
<div class="quote-controls"></div>
 AleksIvanovSinglet:</div>
<blockquote>
<p>Would it be possible for the early access to be granted on a GitHub Org level, so my three colleagues and I can play with the new toys, together ? :slight_smile:</p>
</blockquote>
</aside>
<p>I will check just to be 100% sure, but I think we can do that. :+1:<br>
Is <a href="https://github.com/pipeline-foundation" class="inline-onebox">Pipeline Foundation · GitHub</a> the organization?</p>

**Comment by aleks-ivanov** [comment]
<aside class="quote group-github-employees" data-username="bogdanap" data-post="8" data-topic="156428">
<div class="title">
<div class="quote-controls"></div>
 bogdanap:</div>
<blockquote>
<p>Is <a href="https://github.com/pipeline-foundation" rel="noopener nofollow ugc">Pipeline Foundation · GitHub </a> the organization?</p>
</blockquote>
</aside>
<p>Yes :slight_smile: (post must be at least 20 characters)</p>

**Comment by bogdanap** [comment]
<p>Ok, <a class="mention" href="/aleksivanovsinglet">@AleksIvanovSinglet</a>, I have enabled access to the new API features for all repos on <a href="https://github.com/pipeline-foundation" class="inline-onebox">Pipeline Foundation · GitHub</a> and <a href="https://github.com/aleks-ivanov" class="inline-onebox">aleks-ivanov (Aleksandar Ivanov) · GitHub</a>.</p>
<p>:sparkles: You should now be able to see more details in the responses to the <code>/alerts</code>, <code>/alerts/:number</code> and <code>/analyses</code> endpoints.<br>
:sparkles: The <code>/analyses</code> and <code>/alerts</code> endpoints support pagination, and filtering by tool<br>
:sparkles: We have added a new <code>/analyses/:id</code> endpoint that allows you to fetch info for a single analysis.<br>
:sparkles: There is also some more functionality around uploading SARIF files, but I’m not sure if that’s something you’d be interested in.</p>
<p>:bomb: One thing to keep in mind is that we’re planning on removing the <code>instances</code> field from the <code>/alerts/:number</code> response and will be moving that info to a separate endpoint (This is still WIP, but it’s good to be aware of, in case you are using it).</p>
<p>:warning: It’s also important to note that you are in fact receiving confidential information and are bound by our Terms of Service to maintain this confidentiality: <a href="https://docs.github.com/en/github/site-policy/github-terms-of-service#j-beta-previews" class="inline-onebox">GitHub Terms of Service - GitHub Docs</a></p>
<p>I think that should be it for now. Let us know if you encounter any issues or have any questions. :grinning_face_with_smiling_eyes:<br>
We’re looking forward to hearing your feedback!</p>

**Comment by aleks-ivanov** [comment]
<p><a class="mention" href="/bogdanap">@bogdanap</a> that’s awesome! :slight_smile:</p>
<p>We’d be sure to get back to you with feedback!</p>
